test/Exporter/Test/Writer/KnpSnappyWriterTest.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/*
+ * This file is part of the Sonata package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Exporter\Test\Source;
+
+use Exporter\Writer\KnpSnappyWriter;
+
+/**
+ * Tests the KnpSnapyWriter Writer.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+class KnpSnappyWriterTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var string
+     */
+    protected $filename;
+
+    /**
+     * @var string
+     */
+    protected $wkhtmltopdfBinary;
+
+    /**
+     * @var array
+     */
+    protected $snappyOptions;
+
+    /**
+     * @var array
+     */
+    protected $pdfBoxbinary;
+
+    public function setUp()
+    {
+        $this->filename = 'foobar.pdf';
+        $this->snappyOptions = array(
+            'header-html' => '<h1>HEADER HTML</h1>',
+            'footer-html' => '<footer>CUSTOM FOOTER</footer>',
+        );
+        $this->wkhtmltopdfBinary = __DIR__.'/../../../../vendor/h4cc/wkhtmltopdf-amd64/bin/wkhtmltopdf-amd64';
+        $this->pdfBoxBinary =  '/usr/local/bin/pdfbox.jar';
```

Comment:
I use [PdfBox](https://pdfbox.apache.org/1.8/commandline.html) for test the result of Pdf generation.